pkg/controller/keycloak/keycloak_reconciler_test.go

```diff
@@ -54,11 +54,11 @@ func TestKeycloakReconciler_Test_Creating_All(t *testing.T) {
 	//    7) Postgresql Service
 	//    8) Keycloak Service
 	//    9) Keycloak Discovery Service
-	//    10) Keycloak Probe ConfigMap
-	//    11) Keycloak StatefulSets
-	//    12) Keycloak Route
+	//    10) Keycloak Monitoring Service
+	//    11) Keycloak Probe ConfigMap
+	//    12) Keycloak StatefulSets
 	//        Note, that's no MetricsRoute as it needs an established hostname from the root route
-	assert.Equal(t, len(desiredState), 13)
+	assert.Equal(t, len(desiredState), 14)
```

Comment:
Would be nice to have the length of the dseried state defined as a constant. Easier to read, change and maintainable. Single line change in the future.

Reply:
This is one of the problems that we have with no good solution to be honest. 

On one hand, we decided to test the reconciler using unit tests. That decision turned out to work really great. On the other hand, we need additional "safety net" to verify if the reconciler creates all the objects we need in proper order. An alternative would be to use integration tests for, which is even worse.

So to get back to your suggestion - I don't think using a variable in this case makes sense. The variable would be used in only one place - to verify if the number of emitted objects is fine.

Reply:
@slaskawi After a quick look, a top of my mind suggestion would be, create a method to generate/give a `common.DesiredClusterState` that would be expected for the test case. And then compare the expected object (returned by the method) to the one returned by the reconciler. This should eliminate the need for index by index assertion of expected things. Would this be OK? Am i missing something?

Reply:
That's a neat idea. Let me create a JIRA for this.

Reply:
@slaskawi thanks a lot!